packages/forms/src/widgets/DatalistWidget/DatalistWidget.js

```diff
@@ -227,6 +252,28 @@ class DatalistWidget extends React.Component {
 		return [];
 	}
 
+	getDropdownItems(suggestions) {
+		// options with categories
+		if (this.props.options && this.props.options.groupBy) {
+			const category = this.props.options.groupBy;
+			const categoryMap = {};
+			suggestions.forEach((suggestion) => {
+				if (!Object.prototype.hasOwnProperty.call(categoryMap, suggestion[category])) {
+					categoryMap[suggestion[category]] = [{ text: suggestion.label, value: suggestion.value }];
+				} else {
+					categoryMap[suggestion[category]].push({
+						text: suggestion.label,
+						value: suggestion.value,
+					});
+				}
+			});
```

Comment:
I'd rather use a reduce that the forEach. You can pass the initial empty object directly.

Reply:
I did use reduce at first, but eslint warns me that I should not do assignment on accumulator since it's a function paramter...